crates/bevy_transform/src/transform_propagate_system.rs

```diff
@@ -13,7 +14,7 @@ pub fn transform_propagate_system(
         Without<Parent>,
     >,
     mut transform_query: Query<(&Transform, &mut GlobalTransform), With<Parent>>,
-    changed_transform_query: Query<Entity, Changed<Transform>>,
+    changed_transform_query: Query<Entity, Or<(Changed<Transform>, With<DirtyParent>)>>,
```

Comment:
I think `Changed<Children>` would be a better filter here; in that case we could avoid the entire `DirtyParent` bookkeeping. 

I'm not actually sure that (without doing that) any changes to `Children` which didn't also change the child's parent would result in transforms being updated; unless there is a system similar to `update_parents` but the other way around.

Reply:
I did have a discussion about this when implementing it. The options were:

1. Add a marker component
2. Remove the automatic insertion of the `Children` resource, so that it fails in a way which is obvious to the user (how `Transform` and `GlobalTransform` behave)
3. Recalculate when the `Children` change.

My preference was 1. or 2. The reasoning was that `Children` feels like a general `Component` since it's first-class in the API and using it when there are `Parent`/`Child` relationships is encouraged.

IMO, it shouldn't really trigger a `Transform` update; not merely for optimization purposes, but I don't like the idea of the user receiving the `GlobalTransform` update events each time a child is added or removed. I think that would be confusing.

Reply:
I suppose what actually needs to happen is that `Changed<Children>` needs to be reacted to by re-calculating the children, but 'self' doesn't need to be recalculated (unless it already needed to be). The tree has changed, so the transformations of the subtree need updating.

Reply:
I don't quite follow; could you please elaborate?

I think in general `Changed<Children>` is too coarse a filter; under normal operation if a child is changed, there shouldn't be a reason to recalculate the `GlobalTransform` of it's siblings. It's only in this specific case, where we "lose" the insertion since we're in the same stage that we need to react to it.

Reply:
Well my point is: if you add a chilld, how does the child's transform get updated? Neither the parent or child's transforms have changed, so that filter won't activate; so changed will be `false`, so it won't activate.

If you don't want to update 'self''s transform then, the way to implement this would be to add a `changed_children` query.
If this solution is implemented, it cleanly sidesteps the need for the messy `DirtyParent` component.

Reply:
OK I don't understand the solution at all; at least I can't think of a way of doing it without recalculating the siblings which haven't changed.

This is how I think that the algorithm currently works.

1. Go to the root nodes
2. Walk down each node checking if the `Transform` has changed
3. If the `Transform` has changed, then update the `GlobalTransform` _and_ the `GlobalTransform` of the `Children`

In the problematic case, the `Children` has been created in this stage so it doesn't see the child when walking down the tree. In the frame afterwards, neither the parent nor child have an updated `Transform` but has a `Changed<Children>` which would evaluate to `true`.

The `Changed<Children>` query:

1. Can't distinguish _which_ child has changed, which means we'd have to update _all_ the siblings since we need to mark the parent. Note, this is _also_ true of `DirtyParent`, but since it only lasts for a single frame, that doesn't matter since any changes _during_ that frame needs to be calculated anyway.
2. Will also evaluate to true when mutating `Children`.

I cannot see how a single `Changed<Children>` would be able to sidestep both the issues?